gcloud-java-bigquery/src/test/java/com/google/cloud/bigquery/JobTest.java

```diff
@@ -178,12 +189,112 @@ public void testIsDone_NotExists() throws Exception {
+    expect(bigquery.getJob(JOB_INFO.jobId())).andReturn(null);
+    replay(bigquery, mockOptions);
+    initializeJob();
+    assertNull(job.waitFor());
+    verify(mockOptions);
+  }
+
+  @Test
+  public void testWaitForWithCheckingPeriod() throws InterruptedException, TimeoutException {
+    initializeExpectedJob(3);
+    BigQuery.JobOption[] expectedOptions = {BigQuery.JobOption.fields(BigQuery.JobField.STATUS)};
+    TimeUnit timeUnit = createStrictMock(TimeUnit.class);
+    timeUnit.sleep(42);
+    EasyMock.expectLastCall();
+    JobStatus status = createStrictMock(JobStatus.class);
+    expect(status.state()).andReturn(JobStatus.State.RUNNING);
+    expect(status.state()).andReturn(JobStatus.State.DONE);
+    expect(bigquery.options()).andReturn(mockOptions);
+    expect(mockOptions.clock()).andReturn(Clock.defaultClock());
+    Job runningJob = expectedJob.toBuilder().status(status).build();
+    Job completedJob = expectedJob.toBuilder().status(status).build();
+    expect(bigquery.getJob(JOB_INFO.jobId(), expectedOptions)).andReturn(runningJob);
+    expect(bigquery.getJob(JOB_INFO.jobId(), expectedOptions)).andReturn(completedJob);
+    expect(bigquery.getJob(JOB_INFO.jobId())).andReturn(completedJob);
+    replay(status, bigquery, timeUnit, mockOptions);
+    initializeJob();
+    assertSame(completedJob, job.waitFor(WaitForOption.checkEvery(42, timeUnit)));
+    verify(status, timeUnit, mockOptions);
+  }
+
+  @Test
+  public void testWaitForWithCheckingPeriod_Null() throws InterruptedException, TimeoutException {
+    initializeExpectedJob(2);
+    BigQuery.JobOption[] expectedOptions = {BigQuery.JobOption.fields(BigQuery.JobField.STATUS)};
+    TimeUnit timeUnit = createStrictMock(TimeUnit.class);
+    timeUnit.sleep(42);
+    EasyMock.expectLastCall();
+    expect(bigquery.options()).andReturn(mockOptions);
+    expect(mockOptions.clock()).andReturn(Clock.defaultClock());
+    Job runningJob = expectedJob.toBuilder().status(new JobStatus(JobStatus.State.RUNNING)).build();
+    expect(bigquery.getJob(JOB_INFO.jobId(), expectedOptions)).andReturn(runningJob);
+    expect(bigquery.getJob(JOB_INFO.jobId(), expectedOptions)).andReturn(null);
+    expect(bigquery.getJob(JOB_INFO.jobId())).andReturn(null);
+    replay(bigquery, timeUnit, mockOptions);
+    initializeJob();
+    assertNull(job.waitFor(WaitForOption.checkEvery(42, timeUnit)));
+    verify(bigquery, timeUnit, mockOptions);
+  }
+
+  @Test
+  public void testWaitForWithTimeout() throws InterruptedException, TimeoutException {
+    initializeExpectedJob(2);
+    BigQuery.JobOption[] expectedOptions = {BigQuery.JobOption.fields(BigQuery.JobField.STATUS)};
+    TimeUnit timeUnit = createStrictMock(TimeUnit.class);
+    timeUnit.sleep(1);
+    EasyMock.expectLastCall();
+    Clock clock = createStrictMock(Clock.class);
+    EasyMock.expect(clock.millis()).andReturn(0L);
+    EasyMock.expect(clock.millis()).andReturn(1L);
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'EasyMock.expect' due to existing static import 'org.easymock.EasyMock.expect'](https://www.codacy.com/app/mziccard/gcloud-java/file/2653889044/issues/source?bid=3123103&fileBranchId=3307743#l273)